packages/Modal/src/Modal.js

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import extractChildren from "@paprika/helpers/lib/extractChildren";
+import * as styled from "./Modal.styles";
+
+const propTypes = {
+  /** The content for the Modal. */
+  children: PropTypes.node.isRequired,
+
+  /** Control the visibility of the modal */
+  isOpen: PropTypes.bool.isRequired,
+
+  /** Callback triggered when the modal needs to be close */
+  onClose: PropTypes.func,
+
+  /** Callback once the modal has been opened event */
+  onAfterOpen: PropTypes.func,
+
+  /** Callback once the modal has been closed event */
+  onAfterClose: PropTypes.func,
+
+  /* Control the size (max-width) of the modal */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+
+  a11yText: PropTypes.string,
+};
+
+const defaultProps = {
+  onAfterClose: () => {},
+  onClose: () => {},
+  onAfterOpen: () => {},
+  size: ShirtSizes.MEDIUM,
+  a11yText: null,
+};
+
+const Modal = props => {
+  const { isOpen, onClose, onAfterClose, onAfterOpen, size, a11yText, ...moreProps } = props;
+
+  const {
+    "Modal.Overlay": overlayExtracted,
+    "Modal.Header": headerExtracted,
+    "Modal.Content": contentExtracted,
+    "Modal.Footer": footerExtracted,
+    children,
+  } = extractChildren(moreProps.children, ["Modal.Overlay", "Modal.Header", "Modal.Content", "Modal.Footer"]);
+
+  const overlayProps = overlayExtracted ? overlayExtracted.props : {};
+
+  const ariaLabel = a11yText || (headerExtracted ? headerExtracted.props.children : null);
+
+  return (
+    <styled.Overlay
+      isOpen={isOpen}
+      onClose={onClose}
+      onAfterOpen={onAfterOpen}
+      onAfterClose={onAfterClose}
+      {...overlayProps}
+    >
+      {state => (
+        <styled.Wrapper size={size}>
+          <styled.Dialog state={state} role="dialog" aria-modal="true" aria-label={ariaLabel} data-pka-anchor="modal">
```

Comment:
Can we also support role="alertdialog"?

https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/alertdialog.html

The alertdialog role is a special-case dialog role designed specifically for dialogs that divert users' attention to a brief, important message. Its usage is described in the alert dialog design pattern.

Reply:
@oscarkwan You mean add a new property, that controls `role`?

Reply:
Do we need this for an upcoming feature, @oscarkwan?
I think it's a good idea, but easily added in a later PR.  I might want to give a bit of thought to the API.

Reply:
No, not needed for an upcoming feature, it could be added later. Just thought we should also support other dialog roles.

Reply:
I created issue: https://github.com/acl-services/paprika/issues/355

Reply:
Thank you, @DanilAgafonov